hooks/src/index.d.ts

```diff
@@ -106,7 +106,8 @@ export function useCallback<T extends Function>(callback: T, inputs: Inputs): T;
  * This optimization helps to avoid expensive calculations on every render.
  * If no array is provided, a new value will be computed whenever a new function instance is passed as the first argument.
  */
-export function useMemo<T>(factory: () => T, inputs?: Inputs): T;
+// for `inputs`, allow undefined, but don't make it optional as that is very likely a mistake
+export function useMemo<T>(factory: () => T, inputs: Inputs | undefined): T;
```

Comment:
I copied this from [the react types for useMemo](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/86ba65935681b62875e2d5267e8ae9b409935f89/types/react/index.d.ts#L1032)